tla-bmcmt/src/test/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/aux/FakeCtx.scala

```diff
@@ -0,0 +1,14 @@
+package at.forsyte.apalache.tla.bmcmt.stratifiedRules.aux
+
+import at.forsyte.apalache.tla.bmcmt.smt.{SolverConfig, Z3SolverContext}
+import at.forsyte.apalache.tla.lir.TlaEx
+
+/**
+ * A fake context, which just intercepts any expression added via assertGroundExpr and stores it in a Seq.
+ *
+ * We can use it in tests, to observe constraint generation without needing to instrument SMT.
+ */
+class FakeCtx extends Z3SolverContext(SolverConfig.default) {
```

Comment:
In any case, let's call this a mock, I think that's the established lingo for this kind of things.

```suggestion
 * A mock context, which just intercepts any expression added via assertGroundExpr and stores it in a Seq.
 *
 * We can use it in tests, to observe constraint generation without needing to instrument SMT.
 */
class MockZ3SolverContext extends Z3SolverContext(SolverConfig.default) {
```